R/teal_slice.R

```diff
@@ -533,3 +569,34 @@ slices_field <- function(tss, field) {
   checkmate::assert_class(tss, "teal_slices")
   unique(unlist(lapply(tss, function(x) x[[field]])))
 }
+
+#' @param tss `teal_slices` object
+#' @param file `character(1)` specifying path to save to
+#' @export
+#' @rdname teal_slice
+#' @keywords internal
+#'
+slices_store <- function(tss, file) {
+  checkmate::assert_class(tss, "teal_slices")
+  checkmate::assert_path_for_output(file, overwrite = TRUE, extension = "json")
+
+  cat(format(tss, nchar = NULL), "\n", file = file)
+}
+
+#' @param file `character(1)` specifying path to read from
+#' @export
+#' @rdname teal_slice
+#' @keywords internal
+#'
+slices_restore <- function(file) {
+  checkmate::assert_file_exists(file, access = "r", extension = "json")
+
+  tss_j <- jsonlite::fromJSON(file, simplifyDataFrame = FALSE)
+
+  tss_elements <-
+    lapply(tss_j$slices, function(x) {
+      as.teal_slice(Filter(Negate(is.null), x))
+    })
```

Comment:
Remind me why NULLS are dropped here. Formal arguments of `filter_var` that accept NULL always default to NULL, so if slice fields here they are formal arguments to `filter_var`, they either default to NULL anyway or they will raise errors.

Reply:
I actually no longer remember why I put this here. Maybe it was due to the fact that `show_all` was messed up on `format.teal_slices`

Reply:
It is redundant here. Removed in here - d1003716a1501a7cc4791ef9315a9b59d7324078